[WIP][csharp] Duration format property support

balassit opened the pull request:
<!-- Enter details of the change here. Include additional tests that have been done, reference to the issue for tracking, etc. -->
Add support for duration property to be applied to `format`.
Currently only configured/built for c#. Need to determine path forward to implement further.

@wing328 

Fixes #14305 
<!-- Please check the completed items below -->
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ ] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package || exit
  ./bin/generate-samples.sh ./bin/configs/*.yaml || exit
  ./bin/utils/export_docs_generators.sh || exit
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming `7.x.0` minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

devhl-labs commented on modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java:
```diff
@@ -1986,7 +1992,7 @@ protected Set<String> getNullableTypes() {
 
     protected Set<String> getValueTypes() {
-        return new HashSet<>(Arrays.asList("decimal", "bool", "int", "uint", "long", "ulong", "float", "double", "DateTime", "DateOnly", "DateTimeOffset", "Guid"));
+        return new HashSet<>(Arrays.asList("decimal", "bool", "int", "uint", "long", "ulong", "float", "double", "DateTime", "DateOnly", "DateTimeOffset", "TimeSpan", "Guid"));
```
I think this should also be added to getNullableTypes

balassit replied:
The getNullableTypes method mentions it is deprecated and no functionality exists in it currently

devhl-labs replied:
It is deprecated, but it is still in use in some of the derived classes.

balassit replied:
I think we are not on the same page. There is no functionality in the method you mentioned so I am not clear what you are recommending.

```csharp
@Deprecated
    protected Set<String> getNullableTypes() {
        throw new RuntimeException("This method should no longer be used.");
    }
```

devhl-labs replied:
The method gets overridden in some other classes.

balassit commented:
@devhl-labs can you approve?

devhl-labs commented:
I cant maybe @wing328

balassit commented:
> Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/OpenAPITools/openapi-generator/graphs/contributors.
> 
> Let me know if you need help fixing it.
> 
> Ref: https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

Thanks. I think I have now fixed it and rebased all my commits down to one + the sync to merge it from master

wing328 commented:
some builds failed. can you please take a look when you've time?

balassit commented:
@devhl-labs can you help check the generichost failure? I have updated the JsonConverter.mustache, but not sure how to get it to now generate the updates needed to emit as `JsonSerializer.Deserialize<Timespan>`

devhl-labs commented:
Looks like it will need a JsonConverter<TimeSpan> class. Feel free to exclude generichost from this change for now if you like.
